### PR TITLE
Proxy container in elasticsearch pods restarting due to oom-kill during test of scale cluster

### PR DESCRIPTION
**Pull request by mburke5678**
https://bugzilla.redhat.com/show_bug.cgi?id=1860084

### CODE REVIEW

**Comment by mburke5678** [comment]
@vimalk78 PTAL

**Comment by mburke5678** [comment]
@vimalk78  PTAL

File: modules/cluster-logging-deploy-cli.adoc [hunk 1]
```diff
@@ -368,9 +376,11 @@ However, an unmanaged deployment does not receive updates until cluster logging
...
 <5> Specify the number of Elasticsearch nodes. See the note that follows this list.
 <6> Enter the name of an existing storage class for Elasticsearch storage. For best performance, specify a storage class that allocates block storage. If you do not specify a storage class, {product-title} deploys cluster logging with ephemeral storage only.
-<7> Settings for configuring Kibana. Using the CR, you can scale Kibana for redundancy and configure the CPU and memory for your Kibana pods. For more information, see *Configuring Kibana*.
-<8> Settings for configuring the Curator schedule. Curator is used to remove data that is in the Elasticsearch index format prior to {product-title} 4.5 and will be removed in a later release.
-<9> Settings for configuring Fluentd. Using the CR, you can configure Fluentd CPU and memory limits. For more information, see *Configuring Fluentd*.
+<7> Specify the CPU and memory requests for Elasticsearch as needed. The defaults are 16Gi` for the memory request and `1m` for the CPU request.
+<8> Specify the memory requests for the Elasticsearch proxy as needed.The defaults are `64Mi` for the memory request and `100m` for the CPU request.
```
Comment (vimalk78) [inline modules/cluster-logging-deploy-cli.adoc:380]: space after `needed.`

Comment (vimalk78) [inline modules/cluster-logging-deploy-cli.adoc:380]: The defaults for ES proxy are different in different branches
master: 
```
	defaultESProxyMemoryLimit   = "256Mi"
	defaultESProxyMemoryRequest = "256Mi"
```
https://github.com/openshift/elasticsearch-operator/blob/master/internal/k8shandler/defaults.go#L21

release-4.5
```
	defaultESProxyMemoryLimit   = "256Mi"
	defaultESProxyMemoryRequest = "256Mi"
```
https://github.com/openshift/elasticsearch-operator/blob/release-4.6/pkg/k8shandler/defaults.go#L20

release-4.5
```
	defaultESProxyMemoryLimit   = "64Mi"
	defaultESProxyMemoryRequest = "64Mi"
```
https://github.com/openshift/elasticsearch-operator/blob/release-4.5/pkg/k8shandler/defaults.go#L20

File: modules/cluster-logging-deploy-cli.adoc [hunk 2]
```diff
@@ -342,23 +342,31 @@ spec:
...
       storage:
         storageClassName: "<storage-class-name>" <6>
         size: 200G
+      resources: <7>
+        requests:
+          cpu: "1"
+          memory: "16Gi"
+      proxy: <8>
+        limits:
+          memory: 100Mi
```
Comment (vimalk78) [inline modules/cluster-logging-deploy-cli.adoc:355]: the values of proxy memory limit doesnt match with the content below.

File: modules/cluster-logging-deploy-cli.adoc [hunk 3]
```diff
@@ -342,23 +342,31 @@ spec:
...
       audit:
         maxAge: 7d
-    elasticsearch:
       nodeCount: 3 <5>
       storage:
         storageClassName: "<storage-class-name>" <6>
         size: 200G
+      resources: <7>
+        requests:
+          cpu: "1"
```
Comment (vimalk78) [inline modules/cluster-logging-deploy-cli.adoc:351]: what does `1` CPU request mean?
The default CPU request for ES is
```
	defaultESCpuRequest    = "100m"
```
https://github.com/openshift/elasticsearch-operator/blob/release-4.5/pkg/k8shandler/defaults.go#L15

**Review by vimalk78 — COMMENTED** [review]
some comments

File: modules/cluster-logging-deploy-cli.adoc [hunk 4]
```diff
@@ -342,23 +342,31 @@ spec:
...
+          memory: "16Gi"
```
Comment (ewolinetz) [inline modules/cluster-logging-deploy-cli.adoc:352]: in the `cluster-logging-about-crd.adoc` this is `8Gi`, can we update this value to match that?

File: modules/cluster-logging-deploy-cli.adoc (hunk 3 above)
Comment (ewolinetz) [reply modules/cluster-logging-deploy-cli.adoc:351]: can we just omit this line?

File: modules/cluster-logging-deploy-console.adoc [hunk 5]
```diff
@@ -185,18 +185,27 @@ spec:
       storage:
         storageClassName: "<storage-class-name>" <6>
         size: 200G
+        resources: <7>
+          requests:
+            cpu: "1"
+            memory: "16Gi"
```
Comment (ewolinetz) [inline modules/cluster-logging-deploy-console.adoc:191]: similar ask as above

File: modules/cluster-logging-deploy-console.adoc [hunk 6]
```diff
@@ -206,9 +215,12 @@ However, an unmanaged deployment does not receive updates until the cluster logg
...
 <6> Enter the name of an existing storage class for Elasticsearch storage. For best performance, specify a storage class that allocates block storage. If you do not specify a storage class, {product-title} deploys cluster logging with ephemeral storage only.
-<7> Settings for configuring Kibana. Using the CR, you can scale Kibana for redundancy and configure the CPU and memory for your Kibana nodes. For more information, see *Configuring Kibana*.
-<8> Settings for configuring the Curator schedule. Curator is used to remove data that is in the Elasticsearch index format prior to {product-title} 4.5 and will be removed in a later release.
-<9> Settings for configuring Fluentd. Using the CR, you can configure Fluentd CPU and memory limits. For more information, see *Configuring Fluentd*.
+<6> Enter the name of an existing StorageClass for Elasticsearch storage. For best performance, specify a StorageClass that allocates block storage. If you do not specify a StorageClass, {product-title} deploys cluster logging with ephemeral storage only.
+<7> Specify the CPU and memory requests for Elasticsearch as needed. The defaults are 16Gi` for the memory request and `1m` for the CPU request.
```
Comment (ewolinetz) [inline modules/cluster-logging-deploy-console.adoc:219]: `1m` is very very low and likely incorrect as the default CPU request

File: modules/cluster-logging-deploy-cli.adoc [hunk 7]
```diff
@@ -368,9 +376,11 @@ However, an unmanaged deployment does not receive updates until cluster logging
...
 <5> Specify the number of Elasticsearch nodes. See the note that follows this list.
 <6> Enter the name of an existing storage class for Elasticsearch storage. For best performance, specify a storage class that allocates block storage. If you do not specify a storage class, {product-title} deploys cluster logging with ephemeral storage only.
-<7> Settings for configuring Kibana. Using the CR, you can scale Kibana for redundancy and configure the CPU and memory for your Kibana pods. For more information, see *Configuring Kibana*.
-<8> Settings for configuring the Curator schedule. Curator is used to remove data that is in the Elasticsearch index format prior to {product-title} 4.5 and will be removed in a later release.
-<9> Settings for configuring Fluentd. Using the CR, you can configure Fluentd CPU and memory limits. For more information, see *Configuring Fluentd*.
+<7> Specify the CPU and memory requests for Elasticsearch as needed. The defaults are 16Gi` for the memory request and `1m` for the CPU request.
```
Comment (ewolinetz) [inline modules/cluster-logging-deploy-cli.adoc:379]: `1m` is very very low and likely not correct as the default

**Comment by mburke5678** [comment]
@ewolinetz @vimalk78 thanks for the review. I made changes as you suggested.

**Comment by mburke5678** [comment]
@RH-ematysek Would you be appropriate to review this PR, as you reviewed the original dev BZ?

File: modules/cluster-logging-logstore-limits.adoc [hunk 8]
```diff
@@ -39,15 +44,22 @@ spec:
...
+            memory: "64Gi"
+        proxy: <2>
+          limits:
+            memory: 100Mi
+          requests:
+             memory: 100Mi
 ----
-<1> Specify the CPU and memory requests as needed. If you leave these values blank,
+<1> Specify the CPU and memory requests for Elasticsearch as needed. If you leave these values blank,
+the Elasticsearch Operator sets default values that should be sufficient for most deployments. The defaults are `16Gi` for the memory request and `1m` for the CPU request.
```
Comment (ewolinetz) [inline modules/cluster-logging-logstore-limits.adoc:55]: "1m"

**Comment by mburke5678** [comment]
@RH-ematysek PTAL?

**Comment by mburke5678** [comment]
@RH-ematysek PTAL

File: modules/cluster-logging-logstore-limits.adoc (hunk 8 above)
Comment (ewolinetz) [reply modules/cluster-logging-logstore-limits.adoc:55]: @mburke5678 bump on this saying the default is `1m` which is very low

Comment (mburke5678) [reply modules/cluster-logging-logstore-limits.adoc]: @ewolinetz "The defualt CPU request is low and should be changed as needed." ??

Comment (ewolinetz) [reply modules/cluster-logging-logstore-limits.adoc]: no, more that `1m` is a very low value and incorrect... you adjusted it to `1`. We can remove the line "The default CPU request is low and should be changed as needed"

**Comment by mburke5678** [comment]
@anpingli I believe I made the changes as you suggested.

**Review by anpingli — COMMENTED** [review]
/lgtm

File: modules/cluster-logging-deploy-console.adoc [hunk 9]
```diff
@@ -206,9 +214,12 @@ However, an unmanaged deployment does not receive updates until OpenShift Loggin
...
 <5> Specify the number of Elasticsearch nodes. See the note that follows this list.
 <6> Enter the name of an existing storage class for Elasticsearch storage. For best performance, specify a storage class that allocates block storage. If you do not specify a storage class, {product-title} deploys OpenShift Logging with ephemeral storage only.
-<7> Settings for configuring Kibana. Using the CR, you can scale Kibana for redundancy and configure the CPU and memory for your Kibana nodes. For more information, see *Configuring Kibana*.
-<8> Settings for configuring the Curator schedule. Curator is used to remove data that is in the Elasticsearch index format prior to {product-title} 4.5 and will be removed in a later release.
-<9> Settings for configuring Fluentd. Using the CR, you can configure Fluentd CPU and memory limits. For more information, see *Configuring Fluentd*.
+<6> Enter the name of an existing StorageClass for Elasticsearch storage. For best performance, specify a StorageClass that allocates block storage. If you do not specify a StorageClass, {product-title} deploys cluster logging with ephemeral storage only.
```
Comment (bergerhoffer) [inline modules/cluster-logging-deploy-console.adoc:217]: StorageClass -> storage class

File: modules/cluster-logging-logstore-limits.adoc [hunk 10]
```diff
@@ -39,15 +44,21 @@ spec:
...
+            memory: 100Mi
+          requests:
+             memory: 100Mi
 ----
-<1> Specify the CPU and memory requests as needed. If you leave these values blank,
-the Elasticsearch Operator sets default values that should be sufficient for most deployments.
-+
+<1> Specify the CPU and memory requests for Elasticsearch as needed. If you leave these values blank,
+the Elasticsearch Operator sets default values that should be sufficient for most deployments. The default values are`16Gi` for the memory request and `1` for the CPU request.
+<2> Specify the CPU and memory requests for the Elasticsearch proxy as needed. If you leave these values blank, the Elasticsearch Operator sets default values that should be sufficient for most deployments. The default values are`256Mi` for the memory request and `100m` for the CPU request.
```
Comment (bergerhoffer) [inline modules/cluster-logging-logstore-limits.adoc:56]: Not sure if it renders wrong or not, but there should be a space between "are" and "`256Mi`"

**Review by bergerhoffer — COMMENTED** [review]
Just two minor things!

File: modules/cluster-logging-logstore-limits.adoc (hunk 10 above)
Comment (mburke5678) [reply modules/cluster-logging-logstore-limits.adoc]: Last-second change! Thanks for the catch.

**Comment by mburke5678** [comment]
git cherrypick enterprise-4.6

**Comment by mburke5678** [comment]
cherrypick enterprise-4.6

**Pull request merged**